man/systemd-suspend.service.xml

```diff
@@ -117,6 +117,33 @@
     </variablelist>
   </refsect1>
 
+  <refsect1>
+    <title>Environment</title>
+
+    <variablelist class='environment-variables'>
+        <varlistentry>
+          <term><varname>$SYSTEMD_SLEEP_FREEZE_USER_SESSIONS</varname></term>
```

Comment:
Environment variables like this should not be documented in man page, but only in `docs/ENVIRONMENT.md`. They're not treated in the same level as config files/cmdline switch.

Reply:
(And I'd like to invert this switch)

Reply:
Got it will change

Reply:
I inverted the sleep env var, and also renamed the home env var to make clear that we're overriding the default value. I think this is good now

Reply:
Unresolving. Setting `SYSTEMD_HOME_FREEZE_SESSION_OVERRIDE_DEFAULT=false` to actually apply the override seems wrong.

Options:
1) Go back to `SYSTEMD_HOME_FREEZE_SESSION_DEFAULT=false`
2) invert to `SYSTEMD_HOME_SKIP_FREEZE_SESSION=false` (but then we need to make clear somehow that `freezeSession` in the JSON record takes priority over the env var)
3) Like 2, but flip the relationship and make the env var override the setting in the user record

Let me know what seems preferable

Reply:
Hmm, how about introducing a new `LockAllHomesWithFlags` call? Then, conditionalize the `freeze` flag on whether the environment variable for systemd-sleep is set. Only one envvar should be needed with the above approach?

Reply:
Since the override is only useful wrt sleep anyway, having one source of truth sounds better to me.

Reply:
The override is useful outside of sleep. `homectl lock` can ([and will](https://gitlab.gnome.org/Teams/STF/homed/-/issues/4)) happen outside of suspend.

Reply:
I know and I'm not against that. What I'm saying is that probably the only valid case in which you want to disable it is when running into problems about sleeping. Therefore, it makes more sense to disable it in systemd-sleep level alone, instead of changing the behavior of logind.

Reply:
So instead of using another envvar for homed, just allow LockAllHomes to take a flag. Then we can make sd-sleep control the behavior when sleeping, and otherwise we always respect what's set in the user record, i.e. default to true

Reply:
There's nothing preventing the system from going into suspend _after_ a lock. So the user can lock the system via `homectl lock`, then suspend it. In fact, this is the way I think it _should_ work: gnome-shell should make the decision when to evict keys on its own (and I ultimately want to remove the `LockAllHomes` call from sd-sleep). In this case, with your proposal the user session will be incorrectly frozen.

Also, adding a new dbus method, documenting it and the flag, plumbing all of this data through homed and into homework, etc is significantly more complicated than the env var. And all this would be for what is essentially an unsupported hack/workaround. I think I'd much rather tell people who need the workaround to set two environment variables